saleor/dashboard/sites/forms.py

```diff
@@ -8,18 +9,37 @@ class SiteForm(forms.ModelForm):
     class Meta:
         model = Site
         exclude = []
+        labels = {
+            'domain': pgettext_lazy(
+                'Domain name', 'Domain name'),
+            'name': pgettext_lazy(
+                'Display name', 'Display name')}
 
 
 class SiteSettingsForm(forms.ModelForm):
     class Meta:
         model = SiteSettings
         exclude = ['site']
+        labels = {
+            'header_text': pgettext_lazy(
+                'Header text', 'Header text'),
+            'description': pgettext_lazy(
+                'Site description', 'Description')}
```

Comment:
Just "Description" would allow this to share translation with other similar fields.